### DISCUSSION TITLE
False positives

### DISCUSSION THREAD

**Discussion by musicpanda** [WinMerge/winmerge#1570] [General]
After some 3 years of using the same version I finally decided to upgrade to the latest version. 

As far as I can see most is the same. However, while comparing directory trees I noticed false positives. The files are marked as different but when you open them to see the differences it says they are the same. 

I checked one such pair. Both had exactly the same number of bytes.

What is going wrong here? Is there any way to fix this?

**Comment by sdottaka** [comment]
The difference between the results of a folder comparison and a file comparison is usually due to the fact that the compare method is selected other than "Full contents". For example, if compare method is "Modified date and Size", only the modified date and size of the file are compared. Even if the contents of the files are identical, if the modified date and size of the files are different, the files will be considered different in the folder comparison.

Which compare method are you using?
If you are using "Full contents", could you please let us know the files and their current settings so that we can reproduce the problem?

![image](https://user-images.githubusercontent.com/98126/201699806-0df21ae7-f1a1-42c3-894e-65cd6e0b0ca3.png)

**Comment by musicpanda** [comment]
The method used is full contents. The problem is rather rare: less than one in a hundred. 

![wmerge](https://user-images.githubusercontent.com/13930107/201703100-bc9a596c-d30e-4ba3-9687-a3cea20dacde.jpg)

**Comment by musicpanda** [comment]
Here are the properties of two such files that in the directorylist are different but equal when compared as file.
![front1](https://user-images.githubusercontent.com/13930107/201722321-e0c22a35-60b1-429e-a937-4e4c8d29c6cd.jpg)
![front2](https://user-images.githubusercontent.com/13930107/201722326-aa6f2a90-67e7-40db-b6a3-ee847bdbf426.jpg)

**Comment by sdottaka** [comment]
Does this issue always happen with the same pair of files 100% of the time?
Do older versions have this issue?
Could you tell me the version number of the old WinMerge you were using?
It would be helpful if you could provide files and conditions to reproduce the problem.

**Comment by musicpanda** [comment]
I have no idea what the old version number was. The new version has been written over the old one so I guess it is hard to retrieve.

When I redid the same compare (a tree with a few gigabytes of files) this time things went ok. The false positives were no longer positive.

The false positives were normal php files. There was nothing special about them.

**Comment by sdottaka** [comment]
Thank you for trying the comparison again.
I understand that this problem does not always occur.
